pkg/web/apps/apps_psql_repository.go

```diff
@@ -19,16 +22,75 @@ func NewPostgreSQLRepository(db *sql.DB) Repository {
 
 // GetApps retrieves all apps from the database
 func (a *appsPostgreSQLRepository) GetApps() (*[]models.App, error) {
-	app1 := models.App{
-		ID:                    "a0874c82-2b7f-11e9-b210-d663bd873d93",
-		AppID:                 "com.aerogear.app1",
-		AppName:               "app1",
-		NumOfDeployedVersions: 1,
-		NumOfClients:          1,
-		NumOfAppLaunches:      1,
+	rows, err := a.db.Query(`
+	SELECT a.id,a.app_id,a.app_name,
+	COUNT(v.id) as num_of_deployed_versions, 
+	COALESCE(SUM(v.num_of_app_launches),0) as num_of_app_launches,
+	COALESCE(SUM(v.num_of_clients),0) as num_of_clients
```

Comment:
Note that here the number of clients need group by the version_id.
One device/mobile can change the version installed of the same app.

Reply:
I don't fully see why it needs to group by `device_id` and `version_id`? The `version ` table has the number of app launches per version. We can just sum all of these values per app. What is the need to do an extra join to the `device` table?

Reply:
Because the number of clients is the number of mobile devices with the application installed. If I have the app installed than in the database will have the version + my device_id + app_id. However, if I update my device/mobile for a newer version of this app, then in the database will have another row as newer_version + device_id + app_id. In this way, if you do not group by here it will return 2 when in the reality the app is installed just in one mobile device and not in 2. Is it make sense? 

PS>:I understand that the apps/ will return the total per app and not per version.

Reply:
@camilamacedo86 I had a conversation with @davidffrench as he has more insight into this. The purpose of `version.num_of_clients` is to keep track of the number of clients for that specific version. So as it stands the query is retrieving the correct information.